From daa114f48bd55808c5840c4092996a0a6815151b Mon Sep 17 00:00:00 2001
From: Krzysztof Kozlowski <k.kozlowski@samsung.com>
Date: Thu, 27 Jan 2022 21:40:52 +0100
Subject: [PATCH] ARM: dts: Fix booting on Rinato market device The bootloader
 on market Rinato (Gear 2) device checks for revision in compatible field of
 DTB. If it is not present or lower than required then booting fails with:
 "Could not do normal boot. (no DTB found)".

Log of bootloader in case of failure:
h/w: revision  = 0x06
h/w: schematic = SM-R380_Rev0.5_Final_0205
date = 2000/01/01 12:00:21 (UTC)
cmu_div:4, div:5, src_clk:800000000, pixel_clk:31860720
load_kernel: loading boot image from 57344.. (BOOT)
load_kernel: found zImage, size: 0x3267e8
0xbc Error!
Verify_Binary_Signature: failed.
pit_check_signature (BOOT) invalid.
load_kernel: found custom kernel (ret:-2147483647)
 No need to update kernel type.
Detected board: samsung,rinato-rev06
Could not do normal boot. (no DTB found)
: Entering usb mode for SM-R380_EUR_XX (65535)..

Add a "rev06" suffix to compatible to satisfy the bootloader.

Signed-off-by: Krzysztof Kozlowski <k.kozlowski@samsung.com>
---
 arch/arm/boot/dts/exynos3250-rinato.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/exynos3250-rinato.dts b/arch/arm/boot/dts/exynos3250-rinato.dts
index 5f7f8fedf..154f883b5 100644
--- a/arch/arm/boot/dts/exynos3250-rinato.dts
+++ b/arch/arm/boot/dts/exynos3250-rinato.dts
@@ -18,7 +18,7 @@
 
 / {
 	model = "Samsung Rinato board";
-	compatible = "samsung,rinato", "samsung,exynos3250", "samsung,exynos3";
+	compatible = "samsung,rinato", "samsung,exynos3250", "samsung,exynos3", "samsung,rinato-rev06";
 	chassis-type = "watch";
 
 	aliases {
-- 
2.35.0

